src/Button/__tests__/BasicButton.test.tsx

```diff
@@ -1,29 +1,37 @@
 import React from 'react';
-import { render, fireEvent, cleanup } from 'test-utils';
-import { spy } from 'sinon';
+import { render, fireEvent } from 'test-utils';
+import { spy, SinonSpy } from 'sinon';
 import assert from 'assert';
 import BasicButton from '../BasicButton';
 
 describe('Basic Button', function () {
   let getByText;
-  let clickSpy;
+  let clickSpy: SinonSpy;
   beforeEach(function () {
     clickSpy = spy();
     ({ getByText } = render(
-      <BasicButton clickHandler={clickSpy}>
-        Click You
-      </BasicButton>
+      <>
```

Comment:
For unit testing, we should only render and test only thing at a time, especially for things like there where the thing being tested (the `disabled` prop) causes a change in the behavior of the component. In most cases I'd suggest using `context` blocks to set up a separate rendered component where the variable under test is being changed. Doing that, the test suite could be structured as something like this psuedo-code:

```js
describe('BasicButton', function () {
...
  context('when disabled is true', function () {
    beforeEach(function () {
      render(<Button disabled>)
    });
    it('does not fire the click handler', function () {
      ...
    });
  });
  context('when disabled is false', function () {    
    beforeEach(function () {
      render(<Button>)
    });
    it('fires the click handler', function () {
      ...
    });
  });
```
The general idea is that unit tests should always try to isolate the individual _unit_ of code as much as possible. So here, we're rending two buttons side-by-side that call the same handler, and when we try to click the `disabled` button, that handler shouldn't be called. That's simple enough, but it does open the possibility that clicking the non-disabled button will affect the outcome of the test against the disabled one. That's highly unlikely in this this particular test scenario, but we should still always be following best practices. It's good discipline, for one, but also really helpful if we make future changes that increase the complexity of these components.

Reply:
Thanks for pointing this out!! That makes a lot of sense. Fixed through 3c1fde2.